### ISSUE TITLE
Behavior with blank line in itemize or enumerate environment

### ISSUE THREAD

**Issue by atsuyaw** [textlint/textlint-plugin-latex2e#158]
When there is a blank line just before/after comment line in `itemize` or `enumerate` environment, textlint shows error:
```
[Error - 16:15:59] Unexpected node is given. Is the syntax correct?
Error: Unexpected node is given. Is the syntax correct?
```

I encountered this issue with the `tex` file such as the below:
```
\begin{enumerate}
	\item aaaaa

	% \vspace{8zh}
	\item bbbb
\end{enumerate}
```
In case without  comment line or/and blank line, the error is not shown.

I am sorry but I'm not acquainted with programming, so I can't provide any idea for solute this issue.

**Comment by pddg** [comment]
@atsuyaw 
Thank you for reporting the issue.
I reproduced this issue and will work for fixing this.

**Comment by pddg** [comment]
```bash
$ cat << EOF > sample.tex
\begin{enumerate}
  \item aaaaa

  % \vspace{8zh}
  \item bbbb
\end{enumerate}
EOF 
```

<details><summary>npx luparse -lc sample.tex</summary>

```bash
$ npx luparse -lc sample.tex
{
  "kind": "ast.root",
  "content": [
    {
      "kind": "env",
      "name": "enumerate",
      "args": [],
      "content": [
        {
          "kind": "command",
          "name": "item",
          "args": [],
          "location": {
            "start": {
              "offset": 19,
              "line": 2,
              "column": 2
            },
            "end": {
              "offset": 24,
              "line": 2,
              "column": 7
            }
          }
        },
        {
          "kind": "text.string",
          "content": "aaaaa",
          "location": {
            "start": {
              "offset": 25,
              "line": 2,
              "column": 8
            },
            "end": {
              "offset": 30,
              "line": 2,
              "column": 13
            }
          }
        },
        {
          "kind": "parbreak",
          "location": {
            "start": {
              "offset": 30,
              "line": 2,
              "column": 13
            },
            "end": {
              "offset": 49,
              "line": 5,
              "column": 2
            }
          }
        },
        {
          "kind": "command",
          "name": "item",
          "args": [],
          "location": {
            "start": {
              "offset": 49,
              "line": 5,
              "column": 2
            },
            "end": {
              "offset": 54,
              "line": 5,
              "column": 7
            }
          }
        },
        {
          "kind": "text.string",
          "content": "bbbb",
          "location": {
            "start": {
              "offset": 55,
              "line": 5,
              "column": 8
            },
            "end": {
              "offset": 59,
              "line": 5,
              "column": 12
            }
          }
        }
      ],
      "location": {
        "start": {
          "offset": 0,
          "line": 1,
          "column": 1
        },
        "end": {
          "offset": 75,
          "line": 6,
          "column": 16
        }
      }
    }
  ],
  "comment": [
    {
      "kind": "comment",
      "content": " \\vspace{8zh}",
      "location": {
        "start": {
          "offset": 33,
          "line": 4,
          "column": 2
        },
        "end": {
          "offset": 48,
          "line": 5,
          "column": 1
        }
      }
    }
  ]
}
```

</details>

The location of parbreak (next line of `aaaa`) is:

```js
          "location": {
            "start": {
              "offset": 30,
              "line": 2,
              "column": 13
            },
            "end": {
              "offset": 49,
              "line": 5,
              "column": 2
            }
          }
```

The location of comment is:

```js
      "location": {
        "start": {
          "offset": 33,
          "line": 4,
          "column": 2
        },
        "end": {
          "offset": 48,
          "line": 5,
          "column": 1
        }
      }
```

The parbreak seems including the comment node, but it cannot have child nodes actually. So it will become an error.
This might be a issue of latex-utensils. It can be reproduced using the latest version of latex-utensils.

(However, now we use latex-utensils@3.0.0. Even if this is resolved by latex-utensils, we may have to resolve https://github.com/textlint/textlint-plugin-latex2e/pull/79#issuecomment-782787932 first.)

**Comment by pddg** [comment]
```
$ cat << EOF > sample.tex
\begin{document}

% \vspace{8zh}

\end{document}
EOF
```

<details><summary>npx luparse -lc sample.tex</summary>

```json
{
  "kind": "ast.root",
  "content": [
    {
      "kind": "env",
      "name": "document",
      "args": [],
      "content": [
        {
          "kind": "parbreak",
          "location": {
            "start": {
              "offset": 16,
              "line": 1,
              "column": 17
            },
            "end": {
              "offset": 34,
              "line": 5,
              "column": 1
            }
          }
        }
      ],
      "location": {
        "start": {
          "offset": 0,
          "line": 1,
          "column": 1
        },
        "end": {
          "offset": 48,
          "line": 5,
          "column": 15
        }
      }
    },
    {
      "kind": "parbreak",
      "location": {
        "start": {
          "offset": 48,
          "line": 5,
          "column": 15
        },
        "end": {
          "offset": 50,
          "line": 7,
          "column": 1
        }
      }
    }
  ],
  "comment": [
    {
      "kind": "comment",
      "content": " \\vspace{8zh}",
      "location": {
        "start": {
          "offset": 18,
          "line": 3,
          "column": 1
        },
        "end": {
          "offset": 33,
          "line": 4,
          "column": 1
        }
      }
    }
  ]
}
```

</details>

The location of parbreak is:

```js
          "location": {
            "start": {
              "offset": 16,
              "line": 1,
              "column": 17
            },
            "end": {
              "offset": 34,
              "line": 5,
              "column": 1
            }
          }
```

The location of the comment is:

```js
      "location": {
        "start": {
          "offset": 18,
          "line": 3,
          "column": 1
        },
        "end": {
          "offset": 33,
          "line": 4,
          "column": 1
        }
      }
```

In the same way, the parbreak node appears to contain the comment node. However, this can be parsed by tex2tast.

```js
{
  "type": "Document",
  "range": [
    0,
    50
  ],
  "loc": {
    "start": {
      "line": 1,
      "column": 0
    },
    "end": {
      "line": 6,
      "column": 0
    }
  },
  "children": [
    {
      "loc": {
        "start": {
          "line": 3,
          "column": 0
        },
        "end": {
          "line": 4,
          "column": 0
        }
      },
      "range": [
        18,
        33
      ],
      "value": " \\vspace{8zh}",
      "type": "Comment"
    }
  ]
}
```

@tani 
Do you have any ideas as to why a comment just after a blank line in an enumerate environment would be an error?
Is it possible that the following function is affecting this in some way?
https://github.com/textlint/textlint-plugin-latex2e/blob/ab826ee2cbc6916b54b47cbba7b47d39b05272a9/src/latex-to-ast.ts#L525

**Comment by tani** [comment]
Hmm, the code you mentioned is very clear and looks it has no problem. We should also check the code inserting comment nodes. @atsuyaw, Could you give one week to us because the behavior of code is curious.

**Comment by tani** [comment]
We should also report this but to upstream because of the bug @pddg mentioned.

**Comment by atsuyaw** [comment]
@tani 
sure
I can avoid temporarily this issue by comment outing the  blank line.
I just use the blank line for visibility with long `\item` line, so this issue is not serious for me.

@pddg @tani 
Thankyou for your early corresponding.

**Comment by pddg** [comment]
@tani 
We should consider that the `parbreak` node will overlaps the location of comment...
https://github.com/tamuratak/latex-utensils/issues/25#issuecomment-855321363

**Comment by tani** [comment]
I see. Hmm...

**Comment by pddg** [comment]
I found a solution.
Ignore `parbreak` as well as follows.
https://github.com/textlint/textlint-plugin-latex2e/blob/master/src/latex-to-ast.ts#L528-L533
